Issue fixes 1

kfattila opened the pull request:
This PR contains fixes for GitHub issues: 597, 595, 589, 564, 537

bmx8177 commented on src/app/TideIndexApplication.cpp:
```diff
@@ -126,10 +126,11 @@ int TideIndexApplication::main(
   }
   
-  ofstream* out_decoy_fasta = GeneratePeptides::canGenerateDecoyProteins() ?
+/*  TODO: Reocer the option to generate decoy protien fasta file.
```
I'm confused about this TODO. I think the typo is confusing me. What do you want to do with the option?

Also - typo in "protien"

kfattila replied:
I guess, I meant "recover".  The new tide-index does not support an option to produce decoy fasta.

bmx8177 commented on src/util/Params.cpp:
```diff
@@ -821,6 +821,9 @@ Params::Params() : finalized_(false) {
     "neutral loss peaks have an intensity 1/10 of the primary peak. Neutral losses "
     "are not yet implemented for the res-ev score function.",
     "Available for tide-search.", true);
+  InitIntParam("min-precursor-charge", 1, 1, BILLION,
```
It looks like you added the "min-precursor-charge" parameter but did not remove "spectrum-charge" parameter.

kfattila replied:
The spectrum-charge option is still used by other programs in CRUX.

bmx8177 commented on src/app/TideSearchApplication.h:
```diff
@@ -320,7 +318,8 @@ class TideSearchApplication : public CruxApplication {
             proteins(proteins_), locations(locations_), precursor_window(precursor_window_), window_type(window_type_),
             spectrum_min_mz(spectrum_min_mz_), spectrum_max_mz(spectrum_max_mz_), min_scan(min_scan_), max_scan(max_scan_),
-            min_peaks(min_peaks_), search_charge(search_charge_), top_matches(top_matches_), highest_mz(highest_mz_),
+            // min_peaks(min_peaks_), search_charge(search_charge_), top_matches(top_matches_), highest_mz(highest_mz_),
```
I think you can delete this comment.

kfattila replied:
Right.

bmx8177 commented:
It looks like the tide p-value performance test changed a fair bit. Can you remind me what cause this change (I assume a previous merge).

bmx8177 commented on src/app/tide/peptide_mods3.cc:
```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
+// Original class to generate modified peptides. Writes to temporary files
+// before merging them. As the number of possible modifications increases, the
+// number of required temporary files can grow extremely large.
+class ModsOutputter : public IModsOutputter {
```
If we are getting rid of the alternate class. Does it make sense to change the class name to the original version? Before Kaipo edited it this line looked like the below.

"class ModsOutputter {"

kfattila replied:
Removed.

bmx8177 commented on src/app/tide/peptide_mods3.cc:
```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
-
-    ResultMods resultMods(modTable_, modMaxCounts_, maxMods_, peptide, proteins_);
-    double unmodifiedMass = peptide->mass();
+  uint64_t Total() const { return totalWritten_; }
```
Is there a reason you remove the following comment?

"// Return the total number of peptides written"

kfattila replied:
No idea. I put it back.

bmx8177 commented on src/app/tide/peptide_mods3.cc:
```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
-      if (totalWritten_ % 10000000 == 0) {
-        carp(CARP_INFO, "Wrote %lu modified target peptides to temp files", totalWritten_);
+  void InitCountsMapper() {
+ 
```
why is there an empty line here?

kfattila replied:
I removed the empty line.

bmx8177 commented on src/app/tide/peptide_mods3.cc:
```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
-            break;
+  uint64_t totalWritten_;
+
+  //terminal modifications count as a modification and hence
```
add space after "//" and before first character in this comment and next comment.

kfattila replied:
Done.

bmx8177 commented:
@kfattila is there a reason why the performance tests do not include combined p-value? Did we have them before?

bmx8177 commented:
Also do you think it is worth it to put a quick description of what the file is supposed to do at the top of src/app/tide/peptide_mods3.cc? It may make it easier for people in the future to understand the script.

bmx8177 commented on src/app/tide/peptide_mods3.cc:
```diff
@@ -396,15 +378,17 @@ class ModsOutputterAlt : public IModsOutputter {
-
-  map< int, pair<string, RecordWriter*> > tempFiles_; // id -> file, writer (ids must be in ascending order of mass)
-  uint64_t totalWritten_;
+  VariableModTable* mod_table_;
```
why are there variables declared at the beginning and the end of the private class section of ModsOutputter? Doesn't it make more sense for all variables to be together at the beginning of the private section?

kfattila replied:
I moved them to the beginning.

kfattila commented:
"It looks like the tide p-value performance test changed a fair bit. Can you remind me what cause this change (I assume a previous merge)."
I changed the little bit of the null distribution, I think it committed it around January/February. I just updated the performance plot now. The p-value cure actually got better, but somehow percolator's performance decreased on this specific dataset. 

"@kfattila is there a reason why the performance tests do not include combined p-value? Did we have them before?"
I don't know. I did not remove it. I hope. The res-ev test is there.

kfattila commented:
"Also do you think it is worth it to put a quick description of what the file is supposed to do at the top of src/app/tide/peptide_mods3.cc? It may make it easier for people in the future to understand the script."

I added a one sentence description. I think, this code is self-explanatory :)

kfattila commented:
I found another bug, it wrongly printed out some modified decoy peptides. The bug was only in the printing, so it did not affect the performance.
